app/lib/screens/market/buy_tft.dart

```diff
@@ -0,0 +1,717 @@
+    final amountText = amountController.text.trim();
+    final priceText = priceController.text.trim();
+
+    if (amountText.isNotEmpty && priceText.isNotEmpty) {
+      try {
+        final amount = Decimal.parse(amountText);
+        final price = Decimal.parse(priceText);
+        final total = amount * price;
+        totalAmountController.text = roundAmount(total.toString()).toString();
+      } catch (e) {
+        totalAmountController.text = '';
+      }
+    } else {
+      totalAmountController.text = '';
+    }
+  }
+
+  bool _checkForChanges() {
+    final boolean = (amountController.text != widget.offer!.amount) ||
+        (priceController.text != widget.offer!.price);
+    return boolean;
+  }
+
+  _createOrder() async {
+    setState(() {
+      loading = true;
+    });
+    try {
+      final success = await Stellar.createOrder(widget.wallet.stellarSecret,
+          'USDC', 'TFT', amountController.text, priceController.text);
+      if (success) {
+        showDialog(
+          barrierDismissible: false,
+          context: context,
+          builder: (BuildContext context) => CustomDialog(
+              image: Icons.check,
+              title: 'Success!',
+              description: 'Your order was created successfully.',
+              actions: <Widget>[
+                TextButton(
+                  child: const Text('Close'),
+                  onPressed: () {
+                    Navigator.pop(context);
+                    Navigator.pushReplacement(
+                        context,
+                        MaterialPageRoute(
+                          builder: (context) => OrderWidget(
+                            selectedWallet: widget.wallet,
+                          ),
+                        ));
+                    OrderNotifier.emitUpdate();
+                  },
+                )
+              ]),
+        );
+      } else {
+        showDialog(
+          barrierDismissible: false,
+          context: context,
+          builder: (BuildContext context) => CustomDialog(
+              image: Icons.error,
+              title: 'Failed!',
+              type: DialogType.Error,
+              description: 'Failed to create your order.',
+              actions: <Widget>[
+                TextButton(
+                  child: const Text('Close'),
+                  onPressed: () {
+                    Navigator.pop(context);
+                  },
+                )
+              ]),
+        );
+      }
+    } on StellarBalanceException catch (e) {
+      showDialog(
+        barrierDismissible: false,
+        context: context,
+        builder: (BuildContext context) => CustomDialog(
+            image: Icons.error,
+            title: 'Balance Error',
+            type: DialogType.Error,
+            description: e.isLowReserve
+                ? 'You need to fund your account with some XLMs to create your order'
+                : 'Error creating your order: ${e.toString()}',
```

Comment:
Why do we check that? The on-catch block already catches the low balance error only